core/consensus/grandpa/impl/environment_impl.cpp

```diff
@@ -167,10 +167,11 @@ namespace kagome::consensus::grandpa {
         .round = round,
         .message = {.target_hash = vote.hash, .target_number = vote.number}};
     for (const auto &item : justification.items) {
+      BOOST_ASSERT(item.message.type() != typeid(Precommit()));
```

Comment:
Hm... Perhaps it misundestanding.
I expected that:
```suggestion
      BOOST_ASSERT(item.is<Precommit>());
```
and **if** will be removed
```
auto& precommit = boost::relaxed_get<Precommit>(item.message);
```